plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/procedure/FlushJdbcMetadataCacheProcedure.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.jdbc.procedure;
+
+import com.google.common.collect.ImmutableList;
+import io.trino.plugin.jdbc.CachingJdbcClient;
+import io.trino.spi.procedure.Procedure;
+
+import javax.inject.Inject;
+import javax.inject.Provider;
+
+import java.lang.invoke.MethodHandle;
+
+import static io.trino.spi.block.MethodHandleUtil.methodHandle;
+import static java.util.Objects.requireNonNull;
+
+public class FlushJdbcMetadataCacheProcedure
+        implements Provider<Procedure>
+{
+    private static final MethodHandle FLUSH_JDBC_METADATA_CACHE = methodHandle(
+            FlushJdbcMetadataCacheProcedure.class,
+            "flushJdbcMetadataCache");
+
+    private final CachingJdbcClient cachingJdbcClient;
+
+    @Inject
+    public FlushJdbcMetadataCacheProcedure(CachingJdbcClient cachingJdbcClient)
+    {
+        this.cachingJdbcClient = requireNonNull(cachingJdbcClient, "cachingJdbcClient is null");
+    }
+
+    @Override
+    public Procedure get()
+    {
+        return new Procedure(
+                "system",
+                "flush_jdbc_metadata_cache",
+                ImmutableList.of(),
+                FLUSH_JDBC_METADATA_CACHE.bindTo(this));
+    }
+
+    public void flushJdbcMetadataCache()
+    {
+        cachingJdbcClient.flushCache();
```

Comment:
IIRC we can have multiple caching jdbc clients, we always create a per-transaction cache client which is used to enforce consistency.

I'm not sure how this procedure plays with the per-transaction caches but something to test for and be aware of.

Reply:
Hmm i will look into that. But seems it's binded with no intercepting etc.. in case of JDBC connectors.
https://github.com/trinodb/trino/blob/11a7b260e0b948e3fea9a8d871b4744da854c573/plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/JdbcModule.java#L78

Reply:
It's constructed explicitly - not Guice managed. See DefaultJdbcMetadataFactory.

I think you can find it in similar place for Hive.

Reply:
Ok just checked and I think i understand. So its cache over cache. 
We have Global cache which will be cleared by introduced procedure and transaction lvl. cache wich will not be cleared. 
Question is if we want to clear transaction lvl. cache as-well ?
Clearing global cache seams fine as we want to refresh changes which where done outside of Trino. 
From transaction i think it's very rare 🤔  . @findepi @wendigo

Reply:
Transaction level cache MUST NOT be cleared IMO. That's what I meant to say here.

We should have tests to ensure that is the case if possible.

Reply:
Transactionl-level cache is isolated, so it wouldn't be possible to clear it, even if @aczajkowski wanted to

Reply:
@findepi no i don't want to. :)